### PR TITLE
Refactor setTrustline API & handle memos, flags, fee

### PR DESCRIPTION
**Pull request by FlorianBouron**

### CODE REVIEW

File: packages/extension/src/utils/payment.ts [hunk 1]
```diff
@@ -76,11 +76,22 @@ export const buildDestinationTag = (destinationTag: string | undefined): number
   return Number(destinationTag);
 };
 
+export const checkFee = (fee: string | null) => {
```
Comment (FlorianBouron) [inline packages/extension/src/utils/payment.ts:79]: Can you please add this missing unit test?

File: packages/extension/src/utils/format.ts [hunk 2]
```diff
@@ -34,3 +36,13 @@ export const formatToken = (value: number, currency: string = 'XRP', isDrops = f
 
   return `${formatValue(value)} ${currency.toUpperCase()}`;
 };
+
+export const formatFlags = (flags: PaymentFlags | TrustSetFlags) => {
```
Comment (FlorianBouron) [inline packages/extension/src/utils/format.ts:40]: Can you please add the missing unit test?

File: packages/api/src/index.ts [hunk 3]
```diff
@@ -1,4 +1,4 @@
-export * from './addTrustline';
```
Comment (FlorianBouron) [inline packages/api/src/index.ts:1]: Please add back addTrustline

File: packages/api/src/addTrustline/index.ts [hunk 4]
```diff
@@ -1 +0,0 @@
-export * from './addTrustline';
```
[the remark this answers is not in the document]
Comment (FlorianBouron) [reply packages/api/src/addTrustline/index.ts:1]: As discussed, please remove the addTrustline

File: packages/extension/src/utils/payment.ts (hunk 1 above)
Comment (ThibautBremand) [reply packages/extension/src/utils/payment.ts:79]: Done

File: packages/extension/src/utils/format.ts (hunk 2 above)
Comment (ThibautBremand) [reply packages/extension/src/utils/format.ts:40]: Done

File: packages/constants/src/message/message.types.ts [hunk 5]
```diff
@@ -64,7 +64,7 @@ export interface RequestPaymentMessage {
 
 export interface RequestTrustlineMessage {
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/constants/src/message/message.types.ts:65]: For the payment, the interface is called `RequestPaymentMessage `

File: packages/constants/src/message/message.types.ts [hunk 6]
```diff
@@ -18,11 +18,11 @@ import {
 export type RequestMessage =
   | 'SEND_PAYMENT'
   | 'REQUEST_ADDRESS'
-  | 'REQUEST_ADD_TRUSTLINE'
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/constants/src/message/message.types.ts:21]: So after having discussed it, we don't keep it?

File: packages/api/templates/index.html [hunk 7]
```diff
@@ -78,11 +78,29 @@
...
+            fee: '199',
+            flags: {
+              tfClearFreeze: false,
+              tfClearNoRipple: false,
+              tfSetFreeze: true,
+              tfSetNoRipple: true,
+              tfSetfAuth: false
+            }
           };
-          GemWalletApi.addTrustline(transaction)
```
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/api/templates/index.html:85]: Obsolete comment now ?

File: packages/api/src/addTrustline/index.ts (hunk 4 above)
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/api/src/addTrustline/index.ts:1]: Should be done

File: packages/api/templates/index.html (hunk 7 above)
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/api/templates/index.html:85]: Finally we decided to completely remove it, right?

[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/api/templates/index.html:85]: Removed

File: packages/constants/src/message/message.types.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (ThibautBremand) [reply packages/constants/src/message/message.types.ts]: Retrocompatibility should be fine now

File: packages/api/src/index.ts [hunk 8]
```diff
@@ -1,4 +1,4 @@
...
+export * from './setTrustline';
```
Comment (FlorianBouron) [inline packages/api/src/index.ts:1]: Could you please order imports in alphabetical order?

File: packages/api/src/helpers/extensionMessaging.ts [hunk 9]
```diff
@@ -83,7 +84,19 @@ const serializeMessage = (msg: APIMessages): any => {
     modifiedMsg.payload.amount = JSON.stringify(modifiedMsg.payload.amount);
   }
 
-  if (typeof modifiedMsg.payload?.flags === 'object') {
+  if (
+    'payload' in modifiedMsg &&
+    'limitAmount' in modifiedMsg.payload &&
+    typeof modifiedMsg.payload.limitAmount === 'object'
```
Comment (FlorianBouron) [inline packages/api/src/helpers/extensionMessaging.ts:90]: You could replace everything in this if by the following code:
```suggestion
    typeof modifiedMsg.payload?.limitAmount === 'object'
```

File: packages/api/src/helpers/extensionMessaging.ts [hunk 10]
```diff
@@ -83,7 +84,19 @@ const serializeMessage = (msg: APIMessages): any => {
...
+  ) {
+    modifiedMsg.payload.limitAmount = JSON.stringify(modifiedMsg.payload.limitAmount);
+  }
+
+  if (
+    'payload' in modifiedMsg &&
+    'flags' in modifiedMsg.payload &&
+    typeof modifiedMsg.payload.flags === 'object'
```
Comment (FlorianBouron) [inline packages/api/src/helpers/extensionMessaging.ts:98]: You could replace everything in this if by the following code:
```suggestion
    typeof modifiedMsg.payload?.flags === 'object'
```

File: packages/constants/src/event/event.types.ts [hunk 11]
```diff
@@ -4,7 +4,8 @@ import {
   PaymentRequestPayload,
   TrustlineRequestPayload,
```
Comment (FlorianBouron) [inline packages/constants/src/event/event.types.ts:5]: I would rename that type: `SetTrustlineRequestPayload` as we will have other payloads for trustlines in the future.
Same comment on line 8:
`SetTrustlineRequestPayloadDeprecated`

File: packages/api/src/helpers/extensionMessaging.ts (hunk 9 above)
Comment (ThibautBremand) [reply packages/api/src/helpers/extensionMessaging.ts:90]: Done

File: packages/api/src/helpers/extensionMessaging.ts (hunk 10 above)
Comment (ThibautBremand) [reply packages/api/src/helpers/extensionMessaging.ts:98]: Done

File: packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx [hunk 12]
```diff
@@ -164,14 +183,14 @@ export const AddNewTrustline: FC = () => {
...
       setIsParamsMissing(true);
     } else {
-      addTrustline({
-        currency: params.currency,
-        issuer: params.issuer,
+      setTrustline({
+        limitAmount: params.limitAmount,
         fee: params.fee || undefined,
-        value: params.value
+        memos: params.memos ?? undefined,
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx:192]:
```suggestion
        memos: params.memos || undefined,
```

File: packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx [hunk 13]
```diff
@@ -164,14 +183,14 @@ export const AddNewTrustline: FC = () => {
...
+        flags: params.flags ?? undefined
```
Comment (FlorianBouron) [inline packages/extension/src/components/pages/AddNewTrustline/AddNewTrustline.tsx:193]:
```suggestion
        flags: params.flags || undefined
```

**Pull request merged**